### ISSUE TITLE
Wishlist: configure underline behavior wrt descenders

### ISSUE THREAD

**Issue by callegar** [zauguin/luaul#9]
I wonder if it could be possible to make the package also support underlining at a tighter distance from the baseline omitting the descenders. The stackexchange discussion at https://tex.stackexchange.com/questions/36894/underline-omitting-the-descenders considers it using `soul` or `ulem`.

**Comment by zauguin** [comment]
This is unlikely to be implemented. Basically there are multiple issues which make this somewhat out of scope:

  1. `luaul` intentionally tries to reduce it's influence on the text beyond drawing the actual lines whenever possible to avoid any unintentional interference. This makes it very hard to implement a system which strongly depends on the structure of the individual characters.
  2. There are basically four possible implementation for this that I know of, all of them have significant issues:
     * Drawing all characters a second time in bolder and white to hide the underline around them. (This is AFAICT implemented in the linked StackExchange answers) This doesn't work with non white backgrounds (or at least has to be adapted, especially when the background is non uniform), the duplication can lead to issues around text extraction (since everything is present twice) and it's unclear how to handle non-text content (or boxed text content).
     * Enabling fill and stroke for glyph drawing (as done for FakeBold) and setting the stroke color to white. Similar issues as before for colors but no duplication. But it does make glyphs thinner than they are supposed to be which is often unacceptable.
     * Draw all glyphs a second time in a transparency knockout group to make the underline transparent around it. Solves the color issue but leads to insane overhead in the PDF file. Also has problematic interactions with other nodes. Doesn't interact properly with other uses of transparency.
     * Manually calculate intersections of the underline with the glyph outlines and then shorten the line. This is slow, hard to implement, extremely dependent on the font formats involved and breaks when any kind of non-text construct is encountered.

  3. Most of these approaches are pretty independent from the actual underlining and would belong into the glyph drawing code.

So I don't think that there is a reasonable way to implement this in a sufficiently stable and reliable way to fit into lua-ul and therefore should be implemented as separate functionality for now.

**Comment by callegar** [comment]
> Most of these approaches are pretty independent from the actual underlining and would belong into the glyph drawing code.

I was sort of expecting something like that. Is there anything, at least wrt "modern" fonts (OTF) that could allow this to be done at the glyph drawing level? Namely, is there any concept of an "underlined" char that could be used?

**Comment by zauguin** [comment]
> Namely, is there any concept of an "underlined" char that could be used?

No. OpenType fonts contain a suggested height and width for the underline (inherited from Type 1 fonts actually), but that's it. There is a specified flag to mark a font as underlined (for separate underlined fonts), but I'm not aware of any font actually having a separate underlined version.
